clang/lib/Lex/HeaderMap.cpp

```diff
@@ -54,7 +54,9 @@ std::unique_ptr<HeaderMap> HeaderMap::Create(FileEntryRef FE, FileManager &FM) {
   unsigned FileSize = FE.getSize();
   if (FileSize <= sizeof(HMapHeader)) return nullptr;
 
-  auto FileBuffer = FM.getBufferForFile(FE);
+  auto FileBuffer =
+      FM.getBufferForFile(FE, /*IsVolatile=*/false,
+                          /*RequiresNullTerminator=*/true, /*IsText=*/false);
```

Comment:
i think it'd be great to have a test case for this that actually fails (even if only in zOS).

Reply:
So the following tests were previously failing on z/OS due to the binary/text issue and will be fixed by this change
clang/test/Preprocessor/headermap-rel.c
clang/test/Preprocessor/headermap-rel2.c